python/tk_framework_desktopserver/shotgun/scripts/cache_commands.py

```diff
@@ -12,9 +12,27 @@
 import cPickle
 import glob
 import os
+import sqlite3
+import contextlib
 
 def cache(cache_file, data, base_configuration, engine_name, config_data):
-    import sqlite3
+    """
+    Populates the sqlite cache with a row representing the desired pipeline
+    configuration and entity type. If an entry already exists, it is updated.
+
+    :param str cache_file: The path to the sqlite cache file on disk.
+    :param dict data: The raw payload send down by the client.
+    :param str base_configuration: The desired base pipeline configuration's
+        uri.
+    :param str engine_name: The name of the engine to bootstrap into. This
+        is most likely going to be "tk-shotgun"
+    :param dict config_data: All relevant pipeline configuration data. This
+        dict is keyed by pipeline config entity id, each containing a dict
+        that contains, at a minimum, "entity", "lookup_hash", and
+        "contents_hash" keys.
+    """
+    # The local import of sgtk ensures that it occurs after sys.path is set
+    # to what the server sent over.
     import sgtk
```

Comment:
As discussed, this has been refactored such that a single config entity is handled by each call to cache_commands. I've also moved bootstrap routine into its own method, such that sgtk is scoped out afterwards.